### DISCUSSION TITLE
[Feature] Import/bulk upload photos for users without their API key

### DISCUSSION THREAD

**Discussion by Ramblurr** [immich-app/immich#3681] [Feature Request]
### The feature

When I setup a shared immich instance for multiple users, the first thing I want to do after setting up the instance is import/bulk upload the photos of that user. 

Currently I have to ask the user to login (to an empty library), generate an API key and copy/paste it to me. This works, but is complicated for them.

Obviously the user's API keys shouldn't be shared, but I would like to see some admin flag feature for the bulk upload client that lets me choose the user the uploads are for.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by hosaka** [comment]
Wife approval factor can certainly be improved with this! Either I need to explain how to get an API key or login as that user myself, which is less that ideal.

**Comment by bo0tzz** [comment]
If you use oauth with something like authentik you can impersonate a user, login as them without needing their password, and generate an api key.

**Reply by Kladki** [reply]
In my case though, using Kanidm as my provider, there doesn't seem to be a way to impersonate users.
